tests/testthat/test-derive_params_growth_age.R

```diff
@@ -1,41 +1,291 @@
+test_that("derive_params_growth_age Test 1: Weight SDS and percentileworks", {
   vs_data <- tibble::tribble(
     ~USUBJID, ~SEX, ~AGECUR, ~AGEU, ~VSTESTCD, ~VSSTRESN,
-    "1001", "M", 30, "days", "WEIGHT", 10,
-    "1002", "F", 25, "months", "WEIGHT", 100,
+    "1001", "M", 1, "months", "WEIGHT", 5.8,
+    "1002", "F", 517, "days", "WEIGHT", 8.8,
+    "1003", "M", 870, "days", "WEIGHT", 11.3,
+    "1004", "F", 71, "months", "WEIGHT", 27.1,
   )
 
-  fake_meta <- tibble::tribble(
+  # pull WHO metadata for patients <2 years old, CDC metadata for patients >=2 years old
+  meta <- tibble::tribble(
     ~SEX, ~AGE, ~AGEU, ~L, ~M, ~S,
-    "M", 30, "days", 1, 2, 3,
-    "M", 31, "days", 4, 5, 6,
-    "F", 25, "months", 7, 8, 9,
-    "F", 26, "months", 10, 11, 12
+    "M", 30, "days", 0.2303, 4.4525, 0.13413,
+    "F", 517, "days", -0.2561, 10.0196, 0.12305,
+    "M", 870, "days",-0.3310282164, 13.30298997, 0.1088506239,
+    "F", 2161, "days",-1.327138401, 20.03756384, 0.1484897955,
   )
-
   actual <- derive_params_growth_age(
     dataset = vs_data,
     sex = SEX,
     age = AGECUR,
     age_unit = AGEU,
-    meta_criteria = fake_meta,
+    meta_criteria = meta,
     parameter = VSTESTCD == "WEIGHT",
     analysis_var = VSSTRESN,
     set_values_to_sds = exprs(
-      PARAMCD = "WT2AGEZ"
+      PARAMCD = "WTASDS"
+    ),
+    set_values_to_pctl = exprs(
+      PARAMCD = "WTAPCTL"
+    )
+  )
+
+  vs_data_age <- vs_data %>%
+    mutate(AGECUR_c = case_when(
+      AGEU == "days"  ~ AGECUR,
+      AGEU == "months" ~ round(AGECUR*30.4375)),
+      AGEU_c = "days"
+    ) %>%
+    mutate(AGE = AGECUR_c, AGEU = AGEU_c) %>%
+    select(-ends_with("_c"),-AGECUR)
+
+  vs_data_meta <- vs_data_age %>%inner_join(meta, by=c("SEX", "AGE", "AGEU"))
+
+  vs_sds <- vs_data_meta %>%mutate(AVAL = (((VSSTRESN/M)^L)-1)/(L*S))
+  vs_pctl <- vs_sds %>% mutate(AVAL = pnorm(AVAL))
+  expected <- bind_rows(vs_sds, vs_pctl) %>% pull(AVAL)
+
+
+  expect_equal(
+    expected,
+    filter(actual, PARAMCD %in% c("WTASDS", "WTAPCTL")) %>% pull(AVAL)
+  )
+})
+
+## Test 2: Height SDS and percentile works (P50, P97) ----
+test_that("derive_params_growth_age Test 2: Height SDS and percentile works (P50, P97)", {
+  #LL: LENGTH?
+  vs_data <- tibble::tribble(
+    ~USUBJID, ~SEX, ~AGECUR, ~AGEU, ~VSTESTCD, ~VSSTRESN,
+    "1001", "M", 2, "months", "HEIGHT", 58,
+    "1002", "F", 578, "days", "HEIGHT", 87,
+    "1003", "M", 910, "days", "HEIGHT", 92,
+    "1004", "F", 159, "months", "HEIGHT", 170,
+  )
+
+  # pull WHO metadata for patients <2 years old, CDC metadata for patients >=2 years old
+  meta <- tibble::tribble(
+    ~SEX, ~AGE, ~AGEU, ~L, ~M, ~S,
+    "M", 61, "days", 1, 58.4384, 0.03423,
+    "F", 578, "days", 1, 81.6752, 0.03619,
+    "M", 910, "days", 0.2418532658, 90.91041609, 0.0409575918,
+    "F", 4840, "days",1.172254081, 158.2058471, 0.04304528616,
+  )
+
+  actual <- derive_params_growth_age(
+    dataset = vs_data,
```

Comment:
Should the new argument `by_vars` be added there ?

Reply:
good point! same is true for https://github.com/pharmaverse/admiralpeds/pull/54/files. maybe @zdz2101 could make a commit to both these branches to update this after we merge his PR: https://github.com/pharmaverse/admiralpeds/pull/60, as looks like that one is going to need to merge first.

Reply:
Thanks for catching! `by_vars` added to all test codes.